### PR TITLE
[MM-30984] Missing payment email on billing day if no CC is present

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
https://github.com/mattermost/mattermost-server/pull/16351 added an email for when payment fails on the bill date (a webhook received from stripe). This PR expands on that one to add a webhook event type for when no card is added (they're handled differently in stripe/CWS, and email is different than if the payment fails for institutional reasons)

Here's the email:
image

### Testing
Create yourself a user, and promote it to admin (be sure to use an email you can access)
Then, DM me to trigger the email!

Or, run this:
```
curl --location --request POST 'https://mattermost-server-pr-16442.test.mattermost.cloud/api/v4/cloud/webhook' \
--header 'X-Cloud-Token: e1d9yoexe3by9eg49guse4jkyw' \
--header 'Content-Type: application/json' \
--data-raw '{
    "event": "failed-payment-no-card",
    "failed_payment": {}
}'
```

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-30984

#### Release Note
```release-note
In the event a subscription has a payment and there is no card on the account when that payment is processed, an email will be sent to attempt to get the user to pay up.
```

### CODE REVIEW

**Review by stevemudie — APPROVED** [review]
Tested and passed.

File: app/email.go [hunk 1]
```diff
@@ -788,3 +788,22 @@ func (es *EmailService) SendPaymentFailedEmail(email string, locale string, fail
...
+	bodyPage.Props["Title"] = T("api.templates.payment_failed_no_card.title")
+	bodyPage.Props["Info1"] = T("api.templates.payment_failed_no_card.info1")
+	bodyPage.Props["Info3"] = T("api.templates.payment_failed_no_card.info3")
+	bodyPage.Props["Button"] = T("api.templates.payment_failed_no_card.button")
+
+	if err := es.sendMail(email, subject, bodyPage.Render()); err != nil {
+		return false, model.NewAppError("SendPaymentFailedEmail", "api.user.send_password_reset.send.app_error", nil, "err="+err.Message, http.StatusInternalServerError)
+	}
+
+	return true, nil
```
Comment (marianunez) [inline app/email.go:808]: Do we need to return this bool here?

**Review by marianunez — APPROVED** [review]
LGTM, minor refactoring suggestion

**Review by ethervoid — APPROVED** [review]
LGTM!

File: app/email.go (hunk 1 above)
Comment (nickmisasi) [reply app/email.go:808]: Removed.

**Comment by nickmisasi** [comment]
@matthewbirtch need to get this into a cherry pick for release - if you have any issues with the email design, shoot me a DM and I can fix them as part of a bug!

**Comment by matthewbirtch** [comment]
> @matthewbirtch need to get this into a cherry pick for release - if you have any issues with the email design, shoot me a DM and I can fix them as part of a bug!

All good @nickmisasi. I don't see any issues.

**Comment by nickmisasi** [comment]
Thanks @matthewbirtch !

**Pull request merged**